lib/rules/forbid-foreign-prop-types.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Forbid using another component's propTypes
+ * @author Ian Christian Myers
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Constants
+// ------------------------------------------------------------------------------
+
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Forbid using another component\'s propTypes',
+      category: 'Best Practices',
+      recommended: false
+    }
+  },
+
+  create: function(context) {
+
+    return {
+      MemberExpression: function(node) {
+        if (node.property && node.property.type === 'Identifier' && node.property.name === 'propTypes' ||
+            node.property && node.property.type === 'Literal' && node.property.value === 'propTypes') {
+          context.report({
+            node: node.property,
+            message: 'Using another component\'s propTypes is forbidden'
+          });
+        }
+      },
+
+      ObjectPattern: function(node) {
+        var propTypesNode = node.properties.find(function(property) {
```

Comment:
eslint-plugin-react still supports node 0.10, so please use `array.prototype.find` here instead (after this is merged, i'll update #1038 to switch back to `.find`)